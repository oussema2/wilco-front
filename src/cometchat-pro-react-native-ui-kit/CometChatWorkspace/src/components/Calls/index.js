export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
