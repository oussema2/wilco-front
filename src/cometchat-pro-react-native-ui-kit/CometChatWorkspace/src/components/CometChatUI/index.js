/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
