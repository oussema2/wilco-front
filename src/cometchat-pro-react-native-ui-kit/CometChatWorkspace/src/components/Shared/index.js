export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
