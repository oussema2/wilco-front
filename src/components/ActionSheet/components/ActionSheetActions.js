import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
