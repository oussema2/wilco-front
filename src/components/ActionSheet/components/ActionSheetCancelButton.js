import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
