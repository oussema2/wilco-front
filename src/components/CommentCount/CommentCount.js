import React from 'react';
import {
	View, Text, Image, Pressable
} from 'react-native';
import PropTypes from 'prop-types';
import noop from '../../helpers/noop';
import { comment } from '../../assets/icons';
import { styles } from './CommentCount.styles';

const CommentCount = ( { testID, count, onPress } ) => (
	<Pressable onPress={onPress}>
		<View testID={testID} style={styles.commentCountContainer}>
			<Image testID="comment-count-image" style={styles.commentIcon} source={comment} />
			<Text testID="comment-count-text" style={styles.comments}>
				{`${count} ${count === 1 ? 'comment' : 'comments'}`}
			</Text>
		</View>
	</Pressable>
);

CommentCount.propTypes = {
	testID: PropTypes.string,
	count: PropTypes.number,
	onPress: PropTypes.func
};

CommentCount.defaultProps = {
	testID: 'commentCount-component',
	count: 0,
	onPress: noop
};

export default CommentCount;
