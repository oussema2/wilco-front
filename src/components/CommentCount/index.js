export { default as CommentCount } from './CommentCount';
