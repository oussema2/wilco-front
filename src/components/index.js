export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
