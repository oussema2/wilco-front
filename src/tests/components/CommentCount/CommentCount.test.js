import * as React from 'react';
import { render } from '@testing-library/react-native';
import { CommentCount } from '../../../components/CommentCount';

describe( 'CommentCount', () => {
	describe( 'with the default props', () => {
		it( 'renders the CommentCount component correctly', () => {
			const component = render( <CommentCount /> );
			expect( component.queryByTestId( 'commentCount-component' ) ).toBeDefined();
			expect( component.queryByTestId( 'comment-count-image' ) ).toBeDefined();
			expect( component.queryByTestId( 'comment-count-text' ) ).toHaveTextContent( '0 comments' );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with count prop equal to 1', () => {
		it( 'renders the CommentCount component with "1 comment" text', () => {
			const component = render( <CommentCount count={1} /> );
			expect( component.queryByTestId( 'comment-count-text' ) ).toHaveTextContent( '1 comment' );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with count prop greater than 1', () => {
		it( 'renders the CommentCount component with "<count> comments" text', () => {
			const component = render( <CommentCount count={537} /> );
			expect( component.queryByTestId( 'comment-count-text' ) ).toHaveTextContent( '537 comments' );
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with a custom testID', () => {
		it( 'renders the CommentCount component with the given test ID', () => {
			const component = render( <CommentCount testID="customID" /> );
			expect( component.queryByTestId( 'customID' ) ).toBeDefined();
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
