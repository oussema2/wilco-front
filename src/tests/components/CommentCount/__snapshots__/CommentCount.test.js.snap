// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCount with a custom testID renders the CommentCount component with the given test ID 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    testID="customID"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/comment.png",
        }
      }
      style={
        Object {
          "height": 14.58,
          "tintColor": "#5B5B5B",
          "width": 16.67,
        }
      }
      testID="comment-count-image"
    />
    <Text
      style={
        Object {
          "color": "#5B5B5B",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 13,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "marginLeft": 9.67,
        }
      }
      testID="comment-count-text"
    >
      0 comments
    </Text>
  </View>
</View>
`;

exports[`CommentCount with count prop equal to 1 renders the CommentCount component with "1 comment" text 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    testID="commentCount-component"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/comment.png",
        }
      }
      style={
        Object {
          "height": 14.58,
          "tintColor": "#5B5B5B",
          "width": 16.67,
        }
      }
      testID="comment-count-image"
    />
    <Text
      style={
        Object {
          "color": "#5B5B5B",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 13,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "marginLeft": 9.67,
        }
      }
      testID="comment-count-text"
    >
      1 comment
    </Text>
  </View>
</View>
`;

exports[`CommentCount with count prop greater than 1 renders the CommentCount component with "<count> comments" text 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    testID="commentCount-component"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/comment.png",
        }
      }
      style={
        Object {
          "height": 14.58,
          "tintColor": "#5B5B5B",
          "width": 16.67,
        }
      }
      testID="comment-count-image"
    />
    <Text
      style={
        Object {
          "color": "#5B5B5B",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 13,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "marginLeft": 9.67,
        }
      }
      testID="comment-count-text"
    >
      537 comments
    </Text>
  </View>
</View>
`;

exports[`CommentCount with the default props renders the CommentCount component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    testID="commentCount-component"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/comment.png",
        }
      }
      style={
        Object {
          "height": 14.58,
          "tintColor": "#5B5B5B",
          "width": 16.67,
        }
      }
      testID="comment-count-image"
    />
    <Text
      style={
        Object {
          "color": "#5B5B5B",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 13,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "marginLeft": 9.67,
        }
      }
      testID="comment-count-text"
    >
      0 comments
    </Text>
  </View>
</View>
`;
